packages/opensrp-plans/src/helpers/dataLoaders.ts

```diff
@@ -68,3 +85,325 @@ export async function loadSinglePlan(
+ * @param {string[]} initialOrgs - an array of initial (existing) organization ids
+ * @param {Assignment[]} existingAssignments - an array of Assignment objects that exist for this plan/jurisdiction
+ * @returns {Assignment[]} -
+ */
+export const getAllJursPayload = (
+  selectedOrgs: string[],
+  selectedPlan: PlanDefinition,
+  selectedJurisdictions: string[],
+  initialOrgs: string[] = [],
+  existingAssignments: Assignment[] = []
+): Assignment[] => {
+  let allAssignments: Assignment[] = [];
+  selectedJurisdictions.forEach((selectedJurisdiction) => {
+    const selectedJursAssignments = getSingleJurisdictionPayload(
+      selectedOrgs,
+      selectedPlan,
+      selectedJurisdiction,
+      initialOrgs,
+      existingAssignments
+    );
+    allAssignments = [...allAssignments, ...selectedJursAssignments];
+  });
+  return allAssignments;
+};
+
+// put jurisdictions to plan
+
+/**
+ * send plan payload to api
+ *
+ * @param {string} baseURL -  base url of api
+ * @param {PlanDefinition} planPayload - original plan definition
+ * @param {string []} jurisdictionIds -  the ids to attach to plan
+ * @param {boolean} append - whether to add jurisdictions to existing
+ * @param {OpenSRPService} service - the openSRP service
+ * @param {object} fetchPlanCreator - action creator to add plan to store
+ * @returns {object} -
+ */
+export async function putJurisdictionsToPlan(
+  baseURL: string,
+  planPayload: PlanDefinition,
+  jurisdictionIds: string[],
+  append = false,
+  service: typeof OpenSRPService = OpenSRPService,
+  fetchPlanCreator: typeof fetchPlanDefinitions = fetchPlanDefinitions
+) {
+  const serve = new service(OPENSRP_PLANS, baseURL);
+  // can append operations be idempotent
+  let jurisdictions = jurisdictionIds.map((jurisdictionId) => ({ code: jurisdictionId }));
+  if (append) {
+    // add jurisdictions to existing
+    jurisdictions = [...planPayload.jurisdiction, ...jurisdictions];
+  }
+  const uniqJurisdictions = uniqBy(jurisdictions, 'code');
+  const payload = { ...planPayload, jurisdiction: uniqJurisdictions };
+  return serve
+    .update(payload)
+    .then(() => {
+      fetchPlanCreator([payload]);
+    })
+    .catch((err: Error) => {
+      throw err;
+    });
+}
+
+/** post assignments to api
+ *
+ * @param {string} baseURL -  base url of api
+ * @param {Assignment[]} payload - the assignments payload
+ * @param {fetchAssignments} actionCreator -  dispatches assignments to store
+ * @param {OpenSRPService} service - the openSRP service
+ * @param {boolean} overwrite - whether to replace assignments for this plan
+ * @returns {Promise} - promise
+ */
+export const updateAssignments = (
+  baseURL: string,
+  payload: Assignment[],
+  actionCreator: typeof fetchAssignments = fetchAssignments,
+  service: typeof OpenSRPService = OpenSRPService,
+  overwrite = false
+) => {
+  const serve = new service(OPENSRP_POST_ASSIGNMENTS_ENDPOINT, baseURL);
```

Comment:
Can we pass `OPENSRP_POST_ASSIGNMENTS_ENDPOINT` as prop in the function just like `baseURL`